modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -280,6 +273,72 @@ func (p *Processor) upsertPeerNode(e *store.Edge, spanAttr []*v1_common.KeyValue
 	}
 }
 
+// upsertDatabaseRequest handles the logic of adding a database edge on the
+// graph.  If we have a db.name or db.system attribute, we assume this is a
+// database request.  The name of the edge is determined by the following
+// order:
+//
+//	if we have a peer.service, use it as the database ServerService
+//	if we have a server.address, use it as the database ServerService
+//	if we have a db.name, use it as the database ServerService, which is the backwards-compatible behavior
+func (p *Processor) upsertDatabaseRequest(e *store.Edge, resourceAttr []*v1_common.KeyValue, span *v1_trace.Span) {
+	var (
+		isDatabase bool
+
+		// The fallback database name
+		dbName string
+	)
+
+	// Check for db.name first.  The dbName is set initially to maintain backwards compatbility.
+	if name, ok := processor_util.FindAttributeValue(string(semconv.DBNameKey), resourceAttr, span.Attributes); ok {
```

Comment:
The pattern here loops over the span and resource attributes to find specific keys.  This means that if you use the preferred attributes in the spans, then the generator has to do less work.  I'd considered finding all available keys in one pass, but following the existing pattern elsewhere feels simpler.